### PR TITLE
HasProps and misc cleanup

### PR DESCRIPTION
**Pull request by bryevdv**
related to #10950 

This PR:

* Simplifies `Include` logic by removing the auto-magic around prefixes. Now, `prefix=None` by default and if a prefix is desired it must be supplied explicitly. 
* Consolidates prop-adding logic in `has_props` and removes all related code from `descriptors.py`.  Some of this is temporary just to make remaining task 1 below more straightforward. 
* Some docstring, f-string cleanups. Adds `append_docstring` to encapsulate safe docstring concatenation for -00 in one place
* Removed some cruft and old deprecations
* Removed unneeded `with_bases` complication from `.properties()`

---

The remaining things I am interested in doing soon:

1. Remove explicit caching performed in `HasProps.__new__` and instead add memoized query functions to get various subsets of properties from models. 
2. Make `Alias` a proper property that contributes descriptors like any other property (and can also respond to events)

### CODE REVIEW

**Comment by bryevdv** [comment]
@mattpap FYI regarding `Include`, I also removed the docs for inheriting directly from mixins and would like to de-emphasize that option and perhaps even deprecate it. FWIW the example in the docs was already broken in previous releases:
```
In [8]: class Sample(HasProps, FillProps):
   ...:     pass
   ...:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-31e9f698f1d6> in <module>
----> 1 class Sample(HasProps, FillProps):
      2     pass
      3

~/anaconda/envs/230_39/lib/python3.9/site-packages/bokeh/core/has_props.py in __new__(meta_cls, class_name, bases, class_dict)
    169                 class_dict["__doc__"] += _EXAMPLE_TEMPLATE % dict(path=path)
    170
--> 171         return super().__new__(meta_cls, class_name, bases, class_dict)
    172
    173     def __init__(cls, class_name, bases, nmspc):

TypeError: Cannot create a consistent method resolution
order (MRO) for bases HasProps, FillProps
```
Inheriting together with `Model` (not `HasProps`) does still work (in current and prev) though.

**Comment by bryevdv** [comment]
@mattpap also interested if you have any API prefs about task 1) above. It sucks that `.properties()` returns a list of string property names. Ideally it would return a dict mapping names to property instances or similar, and then other accessor methods could just filter that complete list to return whatever the subset of interest if. Then if these functions are memoized all the gorpy explicit caching and "accumulation" can just be jettisoned.

But I don't think we can break `properties()` as there are definitely downstream things that rely on it. Do you have any preferences here around naming or organization? Maybe we could add a `.props` sub-object, then e.g. 
```
m.props.names
m.props.containers
m.props.dataspecs
```
That could also allow lightening the load  of `HasProps` directly.

File: tests/unit/bokeh/test_model.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_exception_for_bad_callbacks(self) -> None:
     def test_with_propname(self) -> None:
         cb = CustomJS(code="")
         m0 = SomeModel()
-        for name in m0.properties(self):
```
Comment (bryevdv) [inline tests/unit/bokeh/test_model.py:57]: Anyone's guys how many years this wrong usage was here

**Comment by mattpap** [comment]
> Maybe we could add a `.props` sub-object, then e.g.

`.properties` could be such object with an additional `__call__` signature to maintain the existing API. In general I would like us to move towards a direction, where models have as few as possible non-property members, so that there's no room for name collisions (ultimately non, which is how dataclasses work btw.).

**Comment by bryevdv** [comment]
@mattpap that seems like a good compromise, I will plan on that for the next round PR. Do you have any further comments here?

**Pull request merged**